js2-mode.el

```diff
@@ -2297,6 +2297,14 @@ Returns nil if there is no enclosing scope node."
               (not (js2-scope-p node))))
   node)
 
+(defun js2-node-get-enclosing-scope-or-class (node)
+  "Return the innermost `js2-scope' or `js-class' node surrounding NODE.
+Returns nil if there is no enclosing scope node."
+  (while (and (setq node (js2-node-parent node))
+              (not (or (js2-scope-p node)
+                        (js2-class-node-p node)))))
```

Comment:
Why accept class nodes here, but not object nodes? `this` can be used in a function that's simply assigned as a properly to some object.